CONTRIBUTING.md

```diff
@@ -0,0 +1,66 @@
+# Contributing to Code Your Future Immersive Go Course
+
+This repository is a permissively licensed, open source work. We welcome contributions, both as corrections/improvements and additions, and recommend filing an issue for discussion before investing too much work.
+
+## Goals
+
+This repository primarily exists to help junior software engineers get experience with the Go programming language, and learn about topics that would be useful as a software engineer working in production.
+
+It is intended to be followed without needing support (i.e. should explain things, ideally via references to high quality existing materials), but requiring real learning and research by those taking it (i.e. it shouldn't spoon-feed every step and every answer). This is important, as working out how to do things, and thinking about design, are important software engineering skills.
+
+## How we teach
+
+We primarily teach via exercises and projects. We describe outcomes, show examples, and give references (either terms to Google, or links to useful material).
+
+Where we think there isn't a useful / accurate / well-targeted existing resource, we may write explanations ourselves, but this should be rare - many excellent people have come before us!
+
+We don't explain every step along the way - working out the steps is an important part of learning. Deciding exactly how many steps to explain is a tricky balance.
+
+## Project structure
+
+Each project has a name, e.g. `http-auth`.
+
+Every project contains at least the following elements, and may have others as needed:
+
+### On `main` in a directory named for the project
+
+1. A README.md containing:
+   1. Learning objectives. A list of things the person doing the project will understand, know, and be able to do after they have completed the project.
+   1. A concise description of the high-level idea of the project - what's it for?
+   1. A decscription of the outcomes desired from the project, and a rough sequence of steps.
+   1. Ideas for extensions to the project.
+1. Any scaffolding or supporting files that may be useful (e.g. files to be served, starter code if getting started isn't the focus of the project).
+
+### On `main` in the README.md
+
+A reference to the project, in order (or at least after prerequisite projects), and a quick description ofi t.
+
+### On a branch named `impl/${project_name}`
+
+1. A copy of the README.md
```

Comment:
I don't think we should have this — it causes rebase issues that have been fnickety to fix, and even if they're not... it would be confusing for people contributing. Will drop this from previous `impl` branches.